ubuntu-advantage-client/azure_cleanup.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+"""Clean up all running Azure resources and resource groups."""
+
+# Copyright 2020 Canonical Ltd.
+# Lucas Moura <lucas.moura@canonical.com>
+import os
+import json
+
+from pycloudlib.azure.util import get_client
+from azure.mgmt.resource import ResourceManagementClient
+
+import argparse
+
+
+CI_DEFAULT_TAG = "uaclient"
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-t", "--tag", dest="tag", action="store", default=CI_DEFAULT_TAG,
+        help=(
+            "Tag used as a prefix to search for resources to be deleted."
+            "Default: {}".format(CI_DEFAULT_TAG))
+    )
+    parser.add_argument(
+        "--client-id", dest="client_id",
+        help="Client id used to access azure api"
+    )
+    parser.add_argument(
+        "--client-secret", dest="client_secret",
+        help="Client secret used to access azure api"
+    )
+    parser.add_argument(
+        "--tenant-id", dest="tenant_id",
+        help="Tenant id used to access azure api"
+    )
+    parser.add_argument(
+        "--subscription-id", dest="subscription_id",
+        help="Subscription id used to access azure api"
+    )
+    parser.add_argument(
+        "--credentials-file", dest="credentials_file",
+        help="""
+            Json file representing the Azure credentials. That file must a be
+            a json dict containing all the necessary credentials to manage
+            Azure resources. To successfuly be used in this script,
+            the json file must have the following keys:  client_id,
+            client_secret, tenant_id, subscription_id.
+            """
+        )
+
+    return parser.parse_args()
+
+
+def clean_azure(tag, client_id, client_secret, tenant_id, subscription_id):
+    """Clean up all running Azure resources and resource groups"""
+    config_dict = {
+        "clientId": client_id,
+        "clientSecret": client_secret,
+        "tenantId": tenant_id,
+        "subscriptionId": subscription_id
+    }
+
+    resource_client = get_client(
+        ResourceManagementClient, config_dict
+    )
+
+    print('# searching for resource groups matching tag {}'.format(tag))
+    for resource_group in resource_client.resource_groups.list():
+        tags = resource_group.tags
+
+        if tags:
+            for tag_value in tags.values():
+                if tag_value.startswith(tag):
+                    resource_group_name = resource_group.name
+                    print('# deleted resource group: {} with tag {}'.format(
+                        resource_group_name, tag))
+                    result = resource_client.resource_groups.delete(
+                        resource_group_name=resource_group_name
+                    )
+                    result.wait()
+                    break
+
+
+def load_azure_config(credentials_file):
+    with open(credentials_file, 'r') as f:
+        return json.load(f)
```

Comment:
I think we might need a mapping here because the typical .azure/credentials.json on a system looks like the following:
```json
{
  "clientId": "da49e81a-<REDACTED>",
  "clientSecret": "<REDACTED>-",
  "subscriptionId": "12aad61c-6de4-<REDACTED>",
  "tenantId": "a0fb9276-6cc8-4e7a-<REDACTED>",
  "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
  "resourceManagerEndpointUrl": "https://management.azure.com/",
  "activeDirectoryGraphResourceId": "https://graph.windows.net/",
  "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
  "galleryEndpointUrl": "https://gallery.azure.com/",
  "managementEndpointUrl": "https://management.core.windows.net/"
}

```

We might need to only sources clientId, clientSecret, tenantId and subscriptionId  and map them to client_id, client_secret, tenant_id and subscription_id.

This way I can run this script as well from my own system which has previously run 'az login` without having to expand all the stored credentials in a different format from what az cli uses.